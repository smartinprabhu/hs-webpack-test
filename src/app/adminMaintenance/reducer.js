const initialState = {
  createCommentInfo: {},
  orderComments: {},
  orderParts: {},
  orderCheckLists: {},
  orderTimeSheets: {},
  priorityGroupsInfo: {},
  stateGroupsInfo: {},
  stateChangeInfo: {},
  typeGroupsInfo: {},
  workordersInfo: {},
  maintenanceInfo: {},
  // workorderCount: null,
  workorderCountErr: null,
  workorderCountLoading: false,
  maintenanceCount: null,
  maintenanceCountErr: null,
  maintenanceCountLoading: false,
  workorderDetails: {},
  maintenanceExportInfo: {},
  workorderFilters: {},
  cleaningWorkorderFilters: {},
  workorderDashboard: {},
  bookingDelete: {},
  woRows: {},
  bulkOrders: {},
  workorder: {},
  workordersCount: {},
};

// eslint-disable-next-line default-param-last
function bookingMaintenanceReducer(state = initialState, action) {
  switch (action.type) {
    case 'GET_WORK_ORDERS_INFO':
      return {
        ...state,
        workordersInfo: (state.workordersInfo, { loading: true, data: null, err: null }),
      };
    case 'GET_WORK_ORDERS_INFO_SUCCESS':
      return {
        ...state,
        workordersInfo: (state.workordersInfo,
        { loading: false, data: action.payload.data, err: null }),
      };
    case 'GET_WORK_ORDERS_INFO_FAILURE':
      return {
        ...state,
        workordersInfo: (state.workordersInfo, { loading: false, err: action.error, data: null }),
      };
    // case 'GET_WORK_ORDERS_COUNT':
    //   return {
    //     ...state,
    //     workorderCountLoading: true,
    //   };
    // case 'GET_WORK_ORDERS_COUNT_SUCCESS':
    //   return {
    //     ...state,
    //     workorderCount: (state.workorderCount, action.payload.data),
    //     workorderCountLoading: false,
    //   };
    // case 'GET_WORK_ORDERS_COUNT_FAILURE':
    //   return {
    //     ...state,
    //     workorderCount: (state.workorderCount, action.error),
    //     workorderCountErr: (state.workorderCountErr, action.error),
    //     workorderCountLoading: false,
    //   };
    case 'CLEAR_MAINTENANCE_INFO':
      return {
        ...state,
        maintenanceInfo: {},
      };
    case 'GET_MAINTENANCE_INFO':
      return {
        ...state,
        maintenanceInfo: (state.maintenanceInfo, { loading: true, data: null, err: null }),
      };
    case 'GET_MAINTENANCE_INFO_SUCCESS':
      return {
        ...state,
        maintenanceInfo: (state.maintenanceInfo,
        { loading: false, data: action.payload.data, err: null }),
      };
    case 'GET_MAINTENANCE_INFO_FAILURE':
      return {
        ...state,
        maintenanceInfo: (state.maintenanceInfo, { loading: false, err: action.error, data: null }),
      };
    case 'GET_MAINTENANCE_COUNT':
      return {
        ...state,
        maintenanceCountLoading: true,
      };
    case 'GET_MAINTENANCE_COUNT_SUCCESS':
      return {
        ...state,
        maintenanceCount: (state.maintenanceCount, action.payload),
        maintenanceCountLoading: false,
      };
    case 'GET_MAINTENANCE_COUNT_FAILURE':
      return {
        ...state,
        maintenanceCount: (state.maintenanceCount, action.error),
        maintenanceCountErr: (state.maintenanceCountErr, action.error),
        maintenanceCountLoading: false,
      };
    case 'GET_WORK_ORDERS':
      return {
        ...state,
        workorder: (state.workorder, { loading: true, data: null, err: null }),
      };
    case 'GET_WORK_ORDERS_SUCCESS':
      return {
        ...state,
        workorder: (state.workorder,
        { loading: false, data: action.payload.data, err: null }),
      };
    case 'GET_WORK_ORDERS_FAILURE':
      return {
        ...state,
        workorder: (state.workorder, { loading: false, err: action.error, data: null }),
      };
    case 'GET_WORK_ORDERS_COUNT':
      return {
        ...state,
        workordersCount: (state.workordersCount, { loading: false, err: null, data: null }),
      };
    case 'GET_WORK_ORDERS_COUNT_SUCCESS':
      return {
        ...state,
        workordersCount: (state.workordersCount, { loading: false, err: null, data: action.payload }),
      };
    case 'GET_WORK_ORDERS_COUNT_FAILURE':
      return {
        ...state,
        workordersCount: (state.workordersCount, { loading: false, err: action.error.data, data: null }),
      };
    case 'GET_WORK_ORDERS_EXPORT_INFO':
      return {
        ...state,
        maintenanceExportInfo: (state.maintenanceExportInfo,
        { loading: true, data: null, err: null }),
      };
    case 'GET_WORK_ORDERS_EXPORT_INFO_SUCCESS':
      return {
        ...state,
        maintenanceExportInfo: (state.maintenanceExportInfo,
        { loading: false, data: action.payload.data, err: null }),
      };
    case 'GET_WORK_ORDERS_EXPORT_INFO_FAILURE':
      return {
        ...state,
        maintenanceExportInfo: (state.maintenanceExportInfo,
        { loading: false, err: action.error, data: null }),
      };
    case 'GET_TEAMS_GROUP_INFO':
      return {
        ...state,
        typeGroupsInfo: (state.typeGroupsInfo, { loading: true, data: null, err: null }),
      };
    case 'GET_TEAMS_GROUP_INFO_SUCCESS':
      return {
        ...state,
        typeGroupsInfo: (state.typeGroupsInfo,
        { loading: false, data: action.payload.data, err: null }),
      };
    case 'GET_TEAMS_GROUP_INFO_FAILURE':
      return {
        ...state,
        typeGroupsInfo: (state.typeGroupsInfo, { loading: false, err: action.error, data: null }),
      };
    case 'GET_STATES_GROUP_INFO':
      return {
        ...state,
        stateGroupsInfo: (state.stateGroupsInfo, { loading: true, data: null, err: null }),
      };
    case 'GET_STATES_GROUP_INFO_SUCCESS':
      return {
        ...state,
        stateGroupsInfo: (state.stateGroupsInfo,
        { loading: false, data: action.payload.data, err: null }),
      };
    case 'GET_STATES_GROUP_INFO_FAILURE':
      return {
        ...state,
        stateGroupsInfo: (state.stateGroupsInfo, { loading: false, err: action.error, data: null }),
      };
    case 'GET_PRIORITIES_GROUP_INFO':
      return {
        ...state,
        priorityGroupsInfo: (state.priorityGroupsInfo, { loading: true, data: null, err: null }),
      };
    case 'GET_PRIORITIES_GROUP_INFO_SUCCESS':
      return {
        ...state,
        priorityGroupsInfo: (state.priorityGroupsInfo, { loading: false, data: action.payload.data, err: null }),
      };
    case 'GET_PRIORITIES_GROUP_INFO_FAILURE':
      return {
        ...state,
        priorityGroupsInfo: (state.priorityGroupsInfo, { loading: false, err: action.error, data: null }),
      };
    case 'GET_ORDER_DETAILS_INFO':
      return {
        ...state,
        workorderDetails: (state.workorderDetails, { loading: true, data: null, err: null }),
      };
    case 'GET_ORDER_DETAILS_INFO_SUCCESS':
      return {
        ...state,
        workorderDetails: (state.workorderDetails, { loading: false, data: action.payload.data, err: null }),
      };
    case 'GET_ORDER_DETAILS_INFO_FAILURE':
      return {
        ...state,
        workorderDetails: (state.workorderDetails, { loading: false, err: action.error, data: null }),
      };
    case 'GET_COMMENTS_INFO':
      return {
        ...state,
        orderComments: (state.orderComments, { loading: true, data: null, err: null }),
      };
    case 'GET_COMMENTS_INFO_SUCCESS':
      return {
        ...state,
        orderComments: (state.orderComments, { loading: false, data: action.payload.data, err: null }),
      };
    case 'GET_COMMENTS_INFO_FAILURE':
      return {
        ...state,
        orderComments: (state.orderComments, { loading: false, err: action.error, data: null }),
      };
    case 'RESET_COMMENTS_INFO':
      return {
        ...state,
        createCommentInfo: (state.createCommentInfo, { loading: false, err: null, data: null }),
      };
    case 'CREATE_COMMENT_INFO':
      return {
        ...state,
        createCommentInfo: (state.createCommentInfo, { loading: true, data: null, err: null }),
      };
    case 'CREATE_COMMENT_INFO_SUCCESS':
      return {
        ...state,
        createCommentInfo: (state.createCommentInfo, { loading: false, data: action.payload.data, err: null }),
      };
    case 'CREATE_COMMENT_INFO_FAILURE':
      return {
        ...state,
        createCommentInfo: (state.createCommentInfo, { loading: false, err: action.error, data: null }),
      };
    case 'GET_STATE_CHANGE_INFO':
      return {
        ...state,
        stateChangeInfo: (state.stateChangeInfo, { loading: true, data: null, err: null }),
      };
    case 'GET_STATE_CHANGE_INFO_SUCCESS':
      return {
        ...state,
        stateChangeInfo: (state.stateChangeInfo, { loading: false, data: action.payload.data, err: null }),
      };
    case 'GET_STATE_CHANGE_INFO_FAILURE':
      return {
        ...state,
        stateChangeInfo: (state.stateChangeInfo, { loading: false, err: action.error, data: null }),
      };
    case 'GET_STATE_RESET_INFO':
      return {
        ...state,
        stateChangeInfo: (state.stateChangeInfo, { loading: false, data: null, err: null }),
      };
    case 'GET_WORKORDERS_DASHBOARD_INFO':
      return {
        ...state,
        workorderDashboard: (state.workorderDashboard, { loading: true, data: null, err: null }),
      };
    case 'GET_WORKORDERS_DASHBOARD_INFO_SUCCESS':
      return {
        ...state,
        workorderDashboard: (state.workorderDashboard, { loading: false, data: action.payload.data, err: null }),
      };
    case 'GET_WORKORDERS_DASHBOARD_INFO_FAILURE':
      return {
        ...state,
        workorderDashboard: (state.workorderDashboard, { loading: false, err: action.error, data: null }),
      };
    case 'CLEAR_CLEANING_WORKORDER_FILTERS':
      return {
        ...state,
        cleaningWorkorderFilters: {},
      };
    case 'GET_PARTS_INFO':
      return {
        ...state,
        orderParts: (state.orderParts, { loading: true, data: null, err: null }),
      };
    case 'GET_PARTS_INFO_SUCCESS':
      return {
        ...state,
        orderParts: (state.orderParts, { loading: false, data: action.payload.data, err: null }),
      };
    case 'GET_PARTS_INFO_FAILURE':
      return {
        ...state,
        orderParts: (state.orderParts, { loading: false, err: action.error, data: null }),
      };
    case 'GET_CHECK_LISTS_INFO':
      return {
        ...state,
        orderCheckLists: (state.orderCheckLists, { loading: true, data: null, err: null }),
      };
    case 'GET_CHECK_LISTS_INFO_SUCCESS':
      return {
        ...state,
        orderCheckLists: (state.orderCheckLists, { loading: false, data: action.payload.data, err: null }),
      };
    case 'GET_CHECK_LISTS_INFO_FAILURE':
      return {
        ...state,
        orderCheckLists: (state.orderCheckLists, { loading: false, err: action.error, data: null }),
      };
    case 'GET_TIMESHEETS_INFO':
      return {
        ...state,
        orderTimeSheets: (state.orderTimeSheets, { loading: true, data: null, err: null }),
      };
    case 'GET_TIMESHEETS_INFO_SUCCESS':
      return {
        ...state,
        orderTimeSheets: (state.orderTimeSheets, { loading: false, data: action.payload.data, err: null }),
      };
    case 'GET_TIMESHEETS_INFO_FAILURE':
      return {
        ...state,
        orderTimeSheets: (state.orderTimeSheets, { loading: false, err: action.error, data: null }),
      };
    case 'CLEAR_BOOKINGS_FILTER_WORKORDER':
      return {
        ...state,
        workorderFilters: {},
      };
    case 'CLEAR_BOOKINGS_WORKORDER_COUNT':
      return {
        ...state,
        maintenanceCount: {},
      };
    case 'GET_FILTER_WORKORDER':
      return {
        ...state,
        workorderFilters: (state.workorderFilters, action.payload),
      };
    case 'GET_FILTER_CLEANING_WORKORDER':
      return {
        ...state,
        cleaningWorkorderFilters: (state.cleaningWorkorderFilters, action.payload),
      };
    case 'GET_ROWS_WO':
      return {
        ...state,
        woRows: (state.woRows, action.payload),
      };
    case 'DELETE_BOOKING':
      return {
        ...state,
        bookingDelete: (state.bookingDelete, { loading: true, data: null, err: null }),
      };
    case 'DELETE_BOOKING_SUCCESS':
      return {
        ...state,
        bookingDelete: (state.bookingDelete, { loading: false, data: action.payload.data, err: null }),
      };
    case 'DELETE_BOOKING_FAILURE':
      return {
        ...state,
        bookingDelete: (state.bookingDelete, { loading: false, err: action.error.data, data: null }),
      };
    case 'SAVE_BULK_ORDERS':
      return {
        ...state,
        bulkOrders: (state.bulkOrders, { loading: true, data: null, err: null }),
      };
    case 'SAVE_BULK_ORDERS_SUCCESS':
      return {
        ...state,
        bulkOrders: (state.bulkOrders, { loading: false, data: action.payload.data, err: null }),
      };
    case 'SAVE_BULK_ORDERS_FAILURE':
      return {
        ...state,
        bulkOrders: (state.bulkOrders, { loading: false, err: action.error.data, data: null }),
      };
    case 'CLEAR_CLEANING':
      return {
        ...state,
        bulkOrders: (state.bulkOrders, { loading: false, data: null, err: null }),
      };
    case 'CLEAR_BOOKING':
      return {
        ...state,
        bookingDelete: (state.bookingDelete, { loading: false, data: null, err: null }),
      };
    default:
      return state;
  }
}

export default bookingMaintenanceReducer;
