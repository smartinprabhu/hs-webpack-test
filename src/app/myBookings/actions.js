import { CALL_API } from '../../middleware/api';
import AuthService from '../util/authService';

export const GET_BOOKINGS = 'GET_BOOKINGS';
export const GET_BOOKINGS_SUCCESS = 'GET_BOOKINGS_SUCCESS';
export const GET_BOOKINGS_FAILURE = 'GET_BOOKINGS_FAILURE';
export const SAVE_BOOKINGLAYOUT_VIEW = 'SAVE_BOOKINGLAYOUT_VIEW';

const accessToken = AuthService.getAccessToken;

export const getBookings = (plannedOut, companyId, employeeId) => ({
  [CALL_API]: {
    endpoint: `booking/list/?domain=[["planned_out",">","${plannedOut}"],["company_id","=",${companyId}],["employee_id","=",${employeeId}]]`,
    types: [GET_BOOKINGS, GET_BOOKINGS_SUCCESS, GET_BOOKINGS_FAILURE],
    method: 'GET',
    payload: accessToken,
  },
});

export function saveBookingsLayoutView(result) {
  return {
    type: SAVE_BOOKINGLAYOUT_VIEW,
    payload: result,
  };
}

export const getBookingsWithinDuration = (fromPlannedOut, toPlannedOut, companyId, employeeId) => ({
  [CALL_API]: {
    endpoint: `booking/list/?domain=[["planned_out",">","${fromPlannedOut}"],["planned_out","<","${toPlannedOut}"],["company_id","=",${companyId}],["employee_id","=",${employeeId}]]`,
    types: [GET_BOOKINGS, GET_BOOKINGS_SUCCESS, GET_BOOKINGS_FAILURE],
    method: 'GET',
    payload: accessToken,
  },
});
