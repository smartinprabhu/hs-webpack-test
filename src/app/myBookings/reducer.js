import { GET_BOOKINGS, GET_BOOKINGS_SUCCESS, GET_BOOKINGS_FAILURE, SAVE_BOOKINGLAYOUT_VIEW } from './actions';

const initialState = {
  bookingsList: {},
  saveBookingLayoutViews: null,
};

// eslint-disable-next-line default-param-last
function myBookingsReducer(state = initialState, action) {
  switch (action.type) {
    case GET_BOOKINGS: return {
      ...state,
    };
    case GET_BOOKINGS_SUCCESS: return {
      ...state,
      bookingsList: action.payload.data,
    };
    case SAVE_BOOKINGLAYOUT_VIEW:
      return {
        ...state,
        saveBookingLayoutViews: (state.saveBookingLayoutViews, action.payload),
      };
    case GET_BOOKINGS_FAILURE: return {
      ...state,
      bookingsList: (state.bookingsList, { err: action.error.data }),
    };
    default: return {
      ...state,
    };
  }
}

export default myBookingsReducer;
