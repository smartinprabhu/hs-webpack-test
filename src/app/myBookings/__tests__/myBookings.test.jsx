import React from 'react';
import { useSelector } from 'react-redux';
import '@testing-library/jest-dom/extend-expect';
import { render, screen, fireEvent } from '@testing-library/react';
import FullCalendar from '@fullcalendar/react';
import 'babel-polyfill';

import MyBookings from '../myBookings';
import * as actions from '../actions';
import getBookingsList, { getScheduledBookingsList } from '../service';
import reducer from '../reducer';

jest.mock('@fullcalendar/react', () => jest.fn());
jest.mock('react-redux', () => ({
  useSelector: jest.fn((fn) => fn()),
  useDispatch: () => jest.fn(),
}));
jest.mock('@fullcalendar/react', () => () => <div />);
jest.mock('@fullcalendar/daygrid', () => jest.fn());
jest.mock('@fullcalendar/interaction', () => jest.fn());
jest.mock('@fullcalendar/timegrid', () => jest.fn());
jest.mock('../../analytics/analyticsView', () => jest.fn());

jest.mock('../../booking/createBooking/workStationTypes', () => () => jest.fn());
const bookingsList = [
  {
    access_status: false,
    actual_in: '',
    actual_out: '',
    book_for: 'myself',
    booking_type: 'individual',
    company: {
      id: 63,
      name: 'Water Mark Building',
    },
    employee: {
      id: 6235,
      name: 'Raja',
    },
    id: 90410,
    is_cancelled: false,
    is_host: false,
    members: [],
    planned_in: '2020-08-31 00:00:00',
    planned_out: '2020-08-31 15:00:00',
    planned_status: 'Regular',
    prescreen_status: true,
    shift: {
      id: 47,
      name: 'A',
    },
    space: {
      name: '#WS3',
    },
    user_defined: true,
    uuid: 'b6975ea1-0fe5-4e76-b82c-bc94e49d0eff',
    vendor: {
      id: 1450,
      name: 'Motivity Labs(Tenant)',
    },
    working_hours: 0.0,
  },
  {
    access_status: false,
    actual_in: '',
    actual_out: '',
    book_for: 'myself',
    booking_type: 'individual',
    company: {
      id: 63,
      name: 'Water Mark Building',
    },
    employee: {
      id: 6235,
      name: 'Raja',
    },
    id: 90410,
    is_cancelled: false,
    is_host: false,
    members: [],
    planned_in: '2020-08-31 00:00:00',
    planned_out: '2020-08-31 15:00:00',
    planned_status: 'Regular',
    prescreen_status: true,
    shift: {
      id: 47,
      name: 'A',
    },
    space: {},
    user_defined: true,
    uuid: 'b6975ea1-0fe5-4e76-b82c-bc94e49d0eff',
    vendor: {
      id: 1450,
      name: 'Motivity Labs(Tenant)',
    },
    working_hours: 0.0,
  },
];

const bookingsList1 = [{
  access_status: false,
  actual_in: '',
  actual_out: '',
  book_for: 'myself',
  booking_type: 'individual',
  company: {
    id: 63,
    name: 'Water Mark Building',
  },
  employee: {
    id: 6235,
    name: 'Raja',
  },
  id: 90410,
  is_cancelled: false,
  is_host: false,
  members: [],
  planned_in: '2020-08-31 00:00:00',
  planned_out: '2020-08-31 15:00:00',
  planned_status: 'Regular',
  prescreen_status: false,
  shift: {
    id: 47,
    name: 'A',
  },
  space: {
    name: '#WS3',
  },
  user_defined: true,
  uuid: 'b6975ea1-0fe5-4e76-b82c-bc94e49d0eff',
  vendor: {
    id: 1450,
    name: 'Motivity Labs(Tenant)',
  },
  working_hours: 0.0,
}];

const configObj = {
  data: {
    access: {
      access_type: 'QR',
      enable_access: true,
      skip_occupy: false,
    },
    prescreen: {
      checklists: [
        {
          id: 1711,
          name: 'Covid-19 Checklist.',
        },
      ],
      enable_prescreen: true,
      prescreen_is_mandatory: false,
      prescreen_period: 1.0,
      require_checklist: true,
    },
  },
  booking: { future_limit_days: 7 },
};

const userInfo = {
  vendor: {
    id: 1450,
    name: 'Motivity Labs(Tenant)',
  },
  employee: {
    id: 9845,
    name: 'Raja',
  },
  company: {
    address: 'Water Mark  Building\n222 W. Las Colinas Blvd.\nSuite 755 East\nIrving TX 75039\nUnited States',
    id: 3,
    name: 'Water Mark  Building',
    timezone: 'US/Central',
  },
};
describe('myBookings testing', () => {
  it('MyBookings snapshot testing ', () => {
    FullCalendar.mockImplementation(({ eventClick, dayCellContent, datesSet }) => {
      function handleEventClick(values) {
        eventClick(values);
      }
      function EventDetail(data) {
        dayCellContent(data);
      }
      EventDetail({ date: '2021-02-04T11:27:59.000Z' });
      function getPreviousData(data) {
        datesSet(data);
      }
      getPreviousData({
        startStr: '2020-12-30T17:15:00+05:30',
        endStr: '2020-12-30T18:00:00+05:30',
      });
      return (
        <div
          eventClick={handleEventClick}
          dayCellContent={EventDetail}
          datesSet={getPreviousData}
          events={bookingsList}
        />
      );
    });
    useSelector.mockImplementation((fn) => fn({
      user: { userInfo },
      config: { configObj },
      myBookings: { bookingsList1 },
      bookingInfo: {},
      preScreening: {
        preScreeningProcess: {},
      },
    }));
    useSelector.mockImplementation((fn) => fn({
      user: { userInfo },
      config: { configObj },
      myBookings: { bookingsList },
    }));
    const { asFragment } = render(<MyBookings />);
    expect(asFragment()).toMatchSnapshot();
  });

  // test for reducers

  const initialState = {
    bookingsList: {},
  };

  it('it should return the initial state', () => {
    expect(reducer(undefined, {})).toEqual(initialState);
  });

  it('it should handle the GET_BOOKINGS', () => {
    expect(reducer(initialState, {
      type: actions.GET_BOOKINGS,
    })).toEqual({
      bookingsList: {},
    });
  });

  it('it should handle the GET_BOOKINGS_FAILURE', () => {
    expect(reducer(initialState, {
      type: actions.GET_BOOKINGS_FAILURE,
      error: { data: 'error' },
    })).toEqual({
      bookingsList: {
        err: 'error',
      },
    });
  });

  it('it should handle the GET_BOOKINGS_SUCCESS', () => {
    expect(reducer(initialState, {
      type: actions.GET_BOOKINGS_SUCCESS,
      payload: { data: bookingsList },
    })).toEqual({
      bookingsList,
    });
  });

  // test for actions

  it('test for getBookings function', () => {
    const getBookings = () => actions.getBookings();
    getBookings();
    expect(actions.getBookings()).toBeDefined();
  });

  // test for services

  it('test for getBookingsList function', async () => {
    const dispatch = jest.fn();
    const spy = jest.spyOn(actions, 'getBookings');
    const result = getBookingsList();
    spy.mockImplementation(() => Promise.resolve());
    await result(dispatch);
    expect(dispatch.mock.calls[0]).toBeDefined();
  });

  it('test for getScheduledBookingsList function', async () => {
    const dispatch = jest.fn();
    const spy = jest.spyOn(actions, 'getBookingsWithinDuration');
    const result = getScheduledBookingsList();
    spy.mockImplementation(() => Promise.resolve());
    await result(dispatch);
    expect(dispatch.mock.calls[0]).toBeDefined();
  });

  it('MyBookings dom testing without calling functions', async () => {
    FullCalendar.mockImplementation(() => (<div />));
    useSelector.mockImplementation((fn) => fn({
      user: { userInfo },
      config: { configObj },
      myBookings: { bookingsList },
      bookingInfo: {},
      preScreening: {
        preScreeningProcess: {},
      },
    }));
    const { container } = render(<MyBookings />);
    expect(container).toBeInTheDocument();
  });

  it('MyBookings dom testing while passing configObj data', async () => {
    let executed = false;
    FullCalendar.mockImplementation(({ eventClick, dayCellContent, datesSet }) => {
      const panelData = {
        el: {
          style: {},
        },
        event: {
          el: {
            style: {
              backgroundColor: '#44ADE3',
            },
          },
          _def: {
            extendedProps: bookingsList[0],
          },
        },
      };
      function handleEventClick(values) {
        if (!executed) {
          eventClick(values);
          executed = true;
        }
      }
      handleEventClick(panelData);
      function EventDetail(data) {
        dayCellContent(data);
      }
      EventDetail({ date: '2021-02-04T11:27:59.000Z' });
      function getPreviousData(data) {
        datesSet(data);
      }
      getPreviousData({
        startStr: '2020-12-30T17:15:00+05:30',
        endStr: '2020-12-30T18:00:00+05:30',
      });
      return (
        <div
          eventClick={handleEventClick}
          dayCellContent={EventDetail}
          datesSet={getPreviousData}
          events={bookingsList}
        />
      );
    });
    useSelector.mockImplementation((fn) => fn({
      user: { userInfo },
      access: { accessInfo: { data: { message: 'success' } } },
      occupy: { occupyResponse: {} },
      release: { releaseInfo: {} },
      bookingInfo: { deleteInfo: {} },
      config: {
        configObj: {
          data: {
            access: {
              access_type: 'QR',
              enable_access: true,
              skip_occupy: false,
            },
            prescreen: {
              checklists: [
                {
                  id: 1711,
                  name: 'Covid-19 Checklist.',
                },
              ],
              enable_prescreen: true,
              prescreen_is_mandatory: false,
              prescreen_period: 1.0,
              require_checklist: true,
            },
          },
        },
      },
      myBookings: { bookingsList1 },
      preScreening: {
        preScreeningProcess: {},
      },
    }));
    const { container } = render(<MyBookings />);
    expect(container).toBeInTheDocument();
  });

  it('MyBookings dom testing while clicking BOOk button', async () => {
    FullCalendar.mockImplementation(() => (<div />));

    useSelector.mockImplementation((fn) => fn({
      user: { userInfo },
      config: { configObj },
      myBookings: { bookingsList1 },
      bookingInfo: {},
      preScreening: {
        preScreeningProcess: {},
      },
    }));
    const { container } = render(<MyBookings />);
    fireEvent.click(screen.getByText('BOOK'));
    expect(container).toBeInTheDocument();
  });
  const workStations = [{
    asset_categ_type: 'room',
    availability_status: true,
    bookings: [],
    child_count: 0,
    employee: { id: 6235, name: 'raja' },
    id: 9485,
    is_parent: false,
    site: {
      planned_in: '2020-10-20 01:45:00',
      planned_out: '2020-10-20 10:15:00',
      id: 28,
    },
    latitude: '23',
    longitude: '25',
    max_occupancy: 0,
    parent: { id: 9558, name: 'Row#8' },
    path_name: 'WMB/WBL/F#10/WS/Z#2/Row#8/WS#96',
    position: { xpos: '', ypos: '' },
    space_name: 'Work Station #96',
    space_number: 'WMB-A-00104',
    space_sub_type: { id: 50, name: 'Workstation Area' },
    space_type: { id: 9, name: 'Area' },
    status: 'Ready',
    is_booking_allowed: true,
  }];
  it('MyBookings dom testing while having no employeeID', async () => {
    FullCalendar.mockImplementation(() => (<div />));
    useSelector.mockImplementation((fn) => fn({
      user: {
        userInfo: {
          company: {
            timezone: 'Asia/Kolkata',
          },
          employee: {},
        },
      },
      workStations,
      config: { configObj: {} },
      myBookings: { bookingsList1 },
      bookingInfo: {
        workStations,
      },
      preScreening: {
        preScreeningProcess: { err: {} },
      },
    }));
    const { container } = render(<MyBookings />);
    fireEvent.click(screen.getByTestId('book'));
    fireEvent.click(screen.getByText('Next'));
    expect(container).toBeInTheDocument();
  });
  it('MyBookings dom testing ', async () => {
    FullCalendar.mockImplementation(() => (<div />));
    useSelector.mockImplementation((fn) => fn({
      user: {
        userInfo: {
          company: {
            timezone: 'Asia/Kolkata',
          },
          employee: {},
        },
      },
      workStations,
      config: { configObj: {} },
      myBookings: { bookingsList1 },
      bookingInfo: {
        workStations,
      },
      preScreening: {
        preScreeningProcess: { err: {} },
      },
    }));
    const { container } = render(<MyBookings />);
    expect(container).toBeInTheDocument();
  });

  it('MyBookings dom testing while executing handleEventClick function', async () => {
    let executed = false;
    FullCalendar.mockImplementation(({ eventClick, dayCellContent, datesSet }) => {
      const panelData = {
        el: {
          style: {},
        },
        event: {
          el: {
            style: {
              backgroundColor: '#44ADE3',
            },
          },
          _def: {
            extendedProps: bookingsList[0],
          },
        },
      };
      function handleEventClick(values) {
        if (!executed) {
          eventClick(values);
          executed = true;
        }
      }
      handleEventClick(panelData);
      function EventDetail(data) {
        dayCellContent(data);
      }
      EventDetail({ date: '2021-02-04T11:27:59.000Z' });
      function getPreviousData(data) {
        datesSet(data);
      }
      getPreviousData({
        startStr: '2020-12-30T17:15:00+05:30',
        endStr: '2020-12-30T18:00:00+05:30',
      });
      return (
        <div
          eventClick={handleEventClick}
          dayCellContent={EventDetail}
          datesSet={getPreviousData}
          events={bookingsList}
        />
      );
    });
    useSelector.mockImplementation((fn) => fn({
      user: { userInfo },
      access: { accessInfo: {} },
      occupy: { occupyResponse: {} },
      release: { releaseInfo: {} },
      bookingInfo: { deleteInfo: {} },
      config: {
        configObj: {
          data: {},
        },
      },
      myBookings: { bookingsList1 },
      preScreening: {
        preScreeningProcess: {},
      },
    }));
    const { container } = render(<MyBookings />);
    fireEvent.click(screen.getByTestId('close'));
    expect(container).toBeInTheDocument();
  });

  it('MyBookings dom testing while having no previousData state value', async () => {
    let executed = false;
    FullCalendar.mockImplementation(({ eventClick, dayCellContent, datesSet }) => {
      function handleEventClick(values) {
        if (!executed) {
          eventClick(values);
          executed = true;
        }
      }
      function EventDetail(data) {
        dayCellContent(data);
      }
      EventDetail({ date: '2021-02-04T11:27:59.000Z' });
      function getPreviousData(data) {
        datesSet(data);
      }
      getPreviousData({
        startStr: '2020-12-30T17:15:00+05:30',
        endStr: '2020-12-30T18:00:00+05:30',
      });
      return (
        <div
          eventClick={handleEventClick}
          dayCellContent={EventDetail}
          datesSet={getPreviousData}
          events={bookingsList}
        />
      );
    });
    useSelector.mockImplementation((fn) => fn({
      user: { userInfo: { vendor: {}, employee: {} } },
      access: { accessInfo: {} },
      occupy: { occupyResponse: {} },
      release: { releaseInfo: {} },
      bookingInfo: { deleteInfo: {} },
      config: {
        configObj: {
          data: {},
        },
      },
      myBookings: { bookingsList1 },
      preScreening: {
        preScreeningProcess: {},
      },
    }));
    const { container } = render(<MyBookings />);
    expect(container).toBeInTheDocument();
  });

  it('MyBookings dom testing while having previousData state value', async () => {
    let executed = 0;
    FullCalendar.mockImplementation(({ eventClick, dayCellContent, datesSet }) => {
      const panelData = {
        el: {
          style: {},
        },
        event: {
          el: {
            style: {
              backgroundColor: '#44ADE3',
            },
          },
          _def: {
            extendedProps: bookingsList[0],
          },
        },
      };
      function handleEventClick(values) {
        if (executed !== 2) {
          eventClick(values);
          executed += 1;
        }
      }
      handleEventClick(panelData);
      function EventDetail(data) {
        dayCellContent(data);
      }
      EventDetail({ date: '' });
      function getPreviousData(data) {
        datesSet(data);
      }
      getPreviousData({
        startStr: '2020-12-30T17',
        endStr: '2020-12-30T18',
      });
      return (
        <div
          eventClick={handleEventClick}
          dayCellContent={EventDetail}
          datesSet={getPreviousData}
          events={bookingsList}
        />
      );
    });
    useSelector.mockImplementation((fn) => fn({
      user: { userInfo },
      access: { accessInfo: {} },
      occupy: { occupyResponse: {} },
      release: { releaseInfo: {} },
      bookingInfo: { deleteInfo: {} },
      config: {
        configObj: {
          data: {},
        },
      },
      myBookings: { bookingsList1 },
      preScreening: {
        preScreeningProcess: {},
      },
    }));
    const { container } = render(<MyBookings />);
    expect(container).toBeInTheDocument();
  });
});
