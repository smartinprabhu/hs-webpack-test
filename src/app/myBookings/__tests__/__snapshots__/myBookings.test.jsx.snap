// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`myBookings testing MyBookings snapshot testing  1`] = `
<DocumentFragment>
  <div>
    <div
      class="row"
    >
      <div
        class="col-sm-12 col-md-12 col-lg-12 col-xl-12"
      >
        <div
          class="h-108 row"
        >
          <div
            class="col-2 col-sm-1 col-md-2 col-lg-1"
          >
            <img
              alt="clock"
              height="70"
              src="[object Object]"
              width="70"
            />
          </div>
          <div
            class="col-8 col-sm-9 col-md-8 col-lg-9"
          >
            <h1
              class="bookings-header"
            >
              My Bookings
            </h1>
            <div
              class="ml-2 light-text"
            >
              Here is your overview of your bookings.
            </div>
          </div>
          <div
            class="mt-4 col-2 col-sm-2 col-md-2 col-lg-2"
          />
        </div>
        <div
          events="[object Object],[object Object]"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
